pkg/envoy/ads/stream.go

```diff
@@ -20,6 +22,42 @@ import (
 	"github.com/openservicemesh/osm/pkg/utils"
 )
 
+// This is the workqueue Job implementation to issue a Send Response to a proxy
+type sendJob struct {
+	// Send Response parametres
+	typeurls  mapset.Set
+	proxy     *envoy.Proxy
+	cfg       configurator.Configurator
+	adsStream *xds_discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer
+	request   *xds_discovery.DiscoveryRequest
+	xdsServer *Server
+
+	// Optional waiter
+	jobDoneChannel chan struct{}
+}
+
+// Run implementation for sendResponse job
+func (tj *sendJob) Run() {
+	err := (*tj.xdsServer).sendResponse(tj.typeurls, tj.proxy, tj.adsStream, tj.request, tj.cfg)
+	if err != nil {
+		log.Error().Err(err).Msgf("Failed to create and send %v update to Envoy with xDS Certificate SerialNumber=%s",
+			tj.typeurls, tj.proxy.GetCertificateCommonName().String())
+	}
+	tj.jobDoneChannel <- struct{}{}
+}
+
+// JobName implementation for this job, for logging purposes
+func (tj *sendJob) JobName() string {
```

Comment:
It would be nice to move this out of `stream.go` and into (perhaps) `pkg/envoy/ads/jobs.go`? (Or whatever file name you prefer to match the functionality)

Reply:
done.